src/hotspot/share/oops/fieldInfo.hpp

```diff
@@ -269,16 +266,34 @@ class FieldInfoStream : AllStatic {
   friend class JavaFieldStream;
   friend class FieldStreamBase;
   friend class ClassFileParser;
+  friend class FieldInfoReader;
+
+ private:
+  // Don't generate the table for small classes at all.
+  static const int SEARCH_TABLE_THRESHOLD = 16;
+
+  static inline int search_table_position_width(int stream_length) {
+    assert(stream_length <= (1 << 24), "stream too long");
```

Comment:
This assert is OK for here and now.  The binary search widget I have in mind would be 32-bit clean, and therefore would have no surprise limits..